src/arrayMethodJoin.js

```diff
@@ -4,8 +4,26 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
```

Comment:
It's better to set the prototype directly instead of using __proto__. Replace [].__proto__.join2 = function(separator = ',') { with Array.prototype.join2 = function(separator = ',') {